flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -212,9 +213,125 @@ private JobGraph createJobGraph() {
                             + "This indicates that non-serializable types (like custom serializers) were registered");
         }
 
+        setVertexDescription();
+
         return jobGraph;
     }
 
+    private void setVertexDescription() {
+        for (Map.Entry<Integer, JobVertex> headOpAndJobVertex : jobVertices.entrySet()) {
+            Integer headOpId = headOpAndJobVertex.getKey();
+            JobVertex vertex = headOpAndJobVertex.getValue();
+            StringBuilder builder = new StringBuilder();
+            switch (streamGraph.getVertexDescriptionMode()) {
+                case CASCADING:
+                    getCascadingDescription(builder, headOpId, headOpId);
+                    break;
+                case TREE:
+                    getTreeDescription(builder, headOpId, headOpId, "", true);
+                    break;
+                default:
+                    throw new IllegalArgumentException(
+                            String.format(
+                                    "Description mode %s not supported",
+                                    streamGraph.getVertexDescriptionMode()));
+            }
+            vertex.setOperatorPrettyName(builder.toString());
+        }
+    }
+
+    private void getCascadingDescription(StringBuilder builder, int headOpId, int currentOpId) {
+        StreamNode node = streamGraph.getStreamNode(currentOpId);
+        builder.append(descWithChainedSourcesInfo(node));
+
+        LinkedList<Integer> chainedOutput = getChainedOutputNodes(headOpId, node);
+        if (chainedOutput.isEmpty()) {
+            return;
+        }
+        builder.append(" -> ");
+
+        boolean multiOutput = chainedOutput.size() > 1;
+        if (multiOutput) {
+            builder.append("(");
+        }
+
+        while (!chainedOutput.isEmpty()) {
+            Integer outputId = chainedOutput.pollFirst();
+            getCascadingDescription(builder, headOpId, outputId);
+            if (chainedOutput.isEmpty() && multiOutput) {
+                builder.append(")");
+            } else if (!chainedOutput.isEmpty()) {
+                builder.append(" , ");
+            }
+        }
+    }
+
+    private LinkedList<Integer> getChainedOutputNodes(int headOpId, StreamNode node) {
+        LinkedList<Integer> chainedOutput = new LinkedList<>();
+        for (StreamEdge edge : node.getOutEdges()) {
+            int targetId = edge.getTargetId();
+            if (chainedConfigs.containsKey(headOpId)
+                    && chainedConfigs.get(headOpId).containsKey(targetId)) {
+                chainedOutput.add(targetId);
+            }
+        }
+        return chainedOutput;
+    }
+
+    private void getTreeDescription(
```

Comment:
Perhaps we could simplify the method a bit ? 

For the recursion, we pass the parameters 

- precedingSpaces: the spaces before prefix and the description
- isLast: if the current op is the last of its parent

Then in the method, we could first append
```
precedingSpaces + (isLast ? "+-" : ":-") + description 
```

Then for each outputs, we could call with 
```
precedingSpaces = precedingSpaces + "   "
isLast = chainedOutput.isEmpty()
```